### ISSUE TITLE
feat(adk-skill): auto-inject progressive disclosure guidance and align resource access with Google ADK

### ISSUE THREAD

**Issue by 1111mp** [zavora-ai/adk-rust#658]
## Summary

Add a request-processing hook to `Toolset` and use it from `adk-skill::SkillToolset` to automatically inject Google ADK-style progressive-disclosure guidance into LLM requests. Align the default resource-access behavior with Google ADK Python, where `load_skill_resource` can read a known skill’s bundled resource without requiring a prior `load_skill` call.

## Motivation

`SkillToolset` currently exposes discovery and loading tools, but does not provide the model with the protocol for using them. Every host application must repeat this guidance in agent instructions, and models may attempt to read `references/` resources without understanding the expected L1/L2/L3 workflow.

The existing default `ActivatedOnly` resource policy also differs from Google ADK Python, whose `load_skill_resource` resolves a skill and reads bundled resources directly. This can produce unnecessary tool errors for otherwise valid resource reads.

## Proposed Solution

1. Add an optional async `Toolset::process_llm_request(...)` hook with a no-op default implementation.
2. Have `LlmAgent` invoke this hook for static and invocation-scoped toolsets before every model call.
3. Add `build_skill_system_instruction(...)` and `DEFAULT_SKILL_SYSTEM_INSTRUCTION` to `adk-skill`, modeled after Google ADK Python’s `_build_skill_system_instruction`.
4. Have `SkillToolset` generate and inject the protocol using its actual available skill tools.
5. Make `ResourceAccessPolicy::GoogleCompatible` the default. Keep `ActivatedOnly` as an explicit stricter opt-in.

```rust
pub trait Toolset: Send + Sync {
    async fn tools(
        &self,
        ctx: Arc<dyn ReadonlyContext>,
    ) -> Result<Vec<Arc<dyn Tool>>>;

    async fn process_llm_request(
        &self,
        ctx: Arc<dyn ReadonlyContext>,
        request: &mut LlmRequest,
    ) -> Result<()> {
        Ok(())
    }
}
```

## Affected Crate(s)

- `adk-core`
- `adk-agent`
- `adk-skill`

## Alternatives Considered

- Require each agent or host application to include a hand-written skill protocol. This duplicates policy and produces inconsistent behavior.
- Eagerly inject all skill instructions and resources into every prompt. This defeats progressive disclosure and increases context usage.
- Keep ActivatedOnly as the default. This is stricter, but diverges from Google ADK Python’s default resource-loading behavior.

## Additional Context

Google ADK Python implements runtime-generated skill guidance in `_build_skill_system_instruction` and allows direct bundled resource reads through `load_skill_resource`:
https://github.com/google/adk-python/blob/main/src/google/adk/tools/skill_toolset.py

**Comment by jkmaina** [comment]
Automatic skill guidance remains useful, but #659 needs revision before this issue can close. Preserve the existing activation-gated resource default, keep policy failures observable as tool errors, and make injected guidance agree with the actual tools and recovery behavior. The PR review records the concrete changes and missing regression coverage.

**Comment by jkmaina** [comment]
The revision in #659 preserves activation-gated access and tool errors, fixes contradictory guidance, and adds the requested coverage. Review is now approved; the issue remains open until the merge queue completes.
